pacsman/pynetdicom_client.py

```diff
@@ -0,0 +1,399 @@
+
+            # separate by patient ID, count studies and get most recent
+            patient_id_to_info = {}
+            for study in uid_to_result.values():
+                patient_id = study.PatientID
+                study_id = study.StudyInstanceUID
+                if id in patient_id_to_info:
+                    if study.StudyDate > patient_id_to_info[patient_id].most_recent_study:
+                        most_recent_study = study.StudyDate
+                    else:
+                        most_recent_study = patient_id_to_info[patient_id].most_recent_study
+
+                    prev_study_ids = patient_id_to_info[patient_id].num_studies
+
+                    info = PatientInfo(first_name=study.PatientName.given_name,
+                                       last_name=study.PatientName.family_name,
+                                       dob=study.PatientBirthDate,
+                                       patient_id=patient_id,
+                                       most_recent_study=most_recent_study,
+                                       study_ids=prev_study_ids + [study_id])
+                else:
+                    info = PatientInfo(first_name=study.PatientName.given_name,
+                                       last_name=study.PatientName.family_name,
+                                       dob=study.PatientBirthDate,
+                                       patient_id=patient_id,
+                                       most_recent_study=study.StudyDate,
+                                       study_ids=[study_id])
+                patient_id_to_info[patient_id] = info
+
+            return list(patient_id_to_info.values())
+
+    def studies_for_patient(self, patient_id):
+        ae = AE(ae_title=self.client_ae, scu_sop_class=QueryRetrieveSOPClassList)
+
+        with association(ae, self.pacs_url, self.pacs_port) as assoc:
+            responses = _call_c_find_patients(assoc, 'PatientID', patient_id)
+
+            study_ids = []
+            for (status, result) in responses:
+                if status.Status not in status_success_or_pending:
+                    raise Exception('Studies C-FIND Failure Response: 0x{0:04x}'.format(status.Status))
+
+                # Some PACS send back empty "Success" responses at the end of the list
+                if hasattr(result, 'StudyInstanceUID'):
+                    study_ids.append(result.StudyInstanceUID)
+
+            return study_ids
+
+    def series_for_study(self, study_id, modality_filter=None):
+        """
+        :param study_id: StudyInstanceUID from PACS
+        :return: SeriesInfo
+        """
+        ae = AE(ae_title=self.client_ae, scu_sop_class=QueryRetrieveSOPClassList)
+
+        with association(ae, self.pacs_url, self.pacs_port) as assoc:
+            dataset = Dataset()
+            dataset.StudyInstanceUID = study_id
+
+            # Filtering modality with 'MR\\CT' doesn't seem to work with pynetdicom
+            dataset.Modality = ''
```

Comment:
I'm just curious--why are these particular tags set while others aren't?

Reply:
These tags define what is returned from the C-FIND (if they're empty) or what values to search for (if they're set to a value).